src/Stratis.Bitcoin/Consensus/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -55,9 +66,57 @@ protected override async Task OnMessageReceivedAsync(INetworkPeer peer, Incoming
                 case GetProvenHeadersPayload getHeaders:
                     await this.ProcessGetHeadersAsync(peer, getHeaders).ConfigureAwait(false);
                     break;
+                case VersionPayload version:
+                    await this.ProcessVersionAsync(peer, version).ConfigureAwait(false);
```

Comment:
Do we need to check if PH is activated?

Reply:
Agreed. We should enable this policy only after the last checkpoint && PH are activated

Reply:
While I agree because we call a method to process this payload, isn't better to don't have logic in this switches?
Even thinking about having a better publisher/subscriber implementation where a behavior can register at specific payloads and not every payload

Reply:
yeah logic should go inside the method agreed